packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -190,6 +563,181 @@ export const logExperimentUpdated = async ({
   return emittedEvent.id;
 };
 
+/**
+ * Deletes an experiment by ID and logs the event for the organization
+ * @param id
+ * @param organization
+ */
+export async function deleteExperimentByIdForOrganization(
+  id: string,
+  organization: OrganizationInterface
+) {
+  try {
+    const previous = await findExperiment({
+      experimentId: id,
+      organizationId: organization.id,
+    });
```

Comment:
I’m pretty sure the experiment controller is already fetching the experiment from Mongo. We should just pas that in here instead of the id.

Reply:
If we pass the experiment in (instead of the id) can't we just remove this call to find the `previous` and just call `await logExperimentDeleted(organization, experiment)` before calling `await ExperimentModel.deleteOne...`?

I'm a little confused about the `previous` nomenclature here, as to me it indicates we're trying to find maybe the experiment other than the one we're trying to delete, which, looking at the `logExperimentDeleted` that is just logging which experiment we're about to delete. I also dropped the await ExperimentModel.deleteOne call into the try/catch block.

Let me know what you think of this revised flow.

```
/**
 * Deletes an experiment by ID and logs the event for the organization
 * @param id
 * @param organization
 */
export async function deleteExperimentByIdForOrganization(
  experiment: ExperimentInterface,
  organization: OrganizationInterface
) {
  try {
    await logExperimentDeleted(organization, experiment);

    await ExperimentModel.deleteOne({
      id: experiment.id,
      organization: organization.id,
    });
  } catch (e) {
    logger.error(e);
  }
}
```